tools/SourceKit/tools/sourcekitd/bin/InProc/sourcekitdInProc.cpp

```diff
@@ -111,14 +111,24 @@ void sourcekitd::set_interrupted_connection_handler(
 // sourcekitd_request_sync
 //===----------------------------------------------------------------------===//
 
+/// Create a new SourceKit request handle. Each call of this method is
+/// guaranteed to return a new, unique handle.
+sourcekitd_request_handle_t create_request_handle(void) {
+  static std::atomic<size_t> handle(1);
+  return reinterpret_cast<sourcekitd_request_handle_t>(handle.fetch_add(1));
```

Comment:
Nit: could be relaxed memory order since we only care about it being unique.

Reply:
Thanks. Good idea. Forgot about the relaxed memory order we use everywhere else.